src/IssueViz.Security.UnitTests/OpenInIDE/Api/OpenInIDERequestHandlerTests.cs

```diff
@@ -19,72 +19,251 @@
+            CheckNotCalled(navigatorMock, storeMock);
+        }
+
+        [TestMethod]
+        public async Task ShowHotspot_DataIsValid_NavigationSucceeded_And_IssueAddedToStore()
+        {
+            // Note: this test needs a viz mock that doesn't have any members mocked using Setup(...).
+            // This is because we don't expect any of the members to be called (and if the mock has any
+            // members mocked then the call to converterMock.VerifyAll() will fail when the library tries
+            // to transitively verify the viz members).
+            var hotspotVizMock = new Mock<IAnalysisIssueVisualization>();
+
+            InitializeStateValidator(ValidRequest, true);
+            SetServerResponse(ValidRequest, ValidServerHotspot);
+            SetConversionResponse(ValidServerHotspot, hotspotVizMock.Object);
+            SetNavigationRespone(hotspotVizMock.Object, true);
+            SetStoreExpectedItem(hotspotVizMock.Object);
+
+            // Act
+            await testSubject.ShowHotspotAsync(ValidRequest)
+                .ConfigureAwait(false);
+
+            CheckInfoBarCleared();
+            CheckInfoBarNotShown();
+            CheckCalled(toolWindowService, stateValidatorMock, serverMock, converterMock, navigatorMock, storeMock);
+            CheckNotCalled(hotspotVizMock); // shouldn't have accessed any of the members
+
+            logger.AssertPartialOutputStringExists(ValidRequest.ServerUrl.ToString(), ValidRequest.ProjectKey, ValidRequest.OrganizationKey, ValidRequest.HotspotKey);
+        }
+
+        [TestMethod]
+        public async Task ShowHotspot_DataIsValid_NavigationFailed_IssueStillAddedToStore()
+        {
+            const string hotspotFilePath = "c:\\xx\\yyy.txt";
+            const int hotspotStartLine = -12345;
+            var hotspotViz = CreateHotspotVisualization(hotspotFilePath, hotspotStartLine);
+
+            InitializeStateValidator(ValidRequest, true);
+            SetServerResponse(ValidRequest, ValidServerHotspot);
+            SetConversionResponse(ValidServerHotspot, hotspotViz);
+            SetNavigationRespone(hotspotViz, false);
+            SetStoreExpectedItem(hotspotViz);
+
+            // Act
+            await testSubject.ShowHotspotAsync(ValidRequest)
+                .ConfigureAwait(false);
+
+            CheckInfoBarCleared();
+            CheckInfoBarShown();
+            CheckCalled(toolWindowService, stateValidatorMock, serverMock, converterMock, navigatorMock, storeMock);
+            logger.AssertPartialOutputStringExists(hotspotFilePath, hotspotStartLine.ToString());
+        }
+
+        private static IAnalysisIssueVisualization CreateHotspotVisualization(string filePath, int startLine)
+        {
+            var vizMock = new Mock<IAnalysisIssueVisualization>();
+            // Note: these properties are not marked as Verifiable() since they won't always be read
```

Comment:
I'm not sure why we need to verify this class at all

Reply:
So we can assert in the navigation failure case that the info about the failed navigation is logged.